### ISSUE TITLE
Source Engine Flood Exploit

### ISSUE THREAD

**Issue by Jpofin** [ValveSoftware/Source-1-Games#2902]
Hello,
as an owner of several source-games I hate the fact that anyone with a specific tool/script can crash your source server. No matter which game by the way.

Theres only one hoster that has protection against these kind of attacks.
It's called "Valve Source Engine flood" or "Valve Source Engine exploit".
It's some sort of DDOS as I know but I haven't found a tool or anything yet.

Would be nice if someone that has more information could drop them here so it's getting fixed soon. This thing is up for years already and valve doesn't care.

**Comment by 1157** [comment]
Yes it would be nice,The only thing that works every time is a auto-restart batch script when the sever "hard crashes" it will auto-restart. The crasher just leaves as it's boring over and over.
Sourcemod has some ddos anti-flood not sure how good that works on this type of attack. Incase you need this batch script here it is just change to your info and rename server_restart.bat

`::=======================::
::  SRCDS Guardian 3.0   ::
::         Mooga         ::
::       SRCDS.com       ::
::=======================::

::=========================================================::
:: Thanks To Black-Sky & Drocona for making SRCDS Guardian ::
:: This script is open source.  Feel free to edit at will. ::
::                                                         ::
:: This script was writen for the use of the srcds.com     ::
:: online community.  If you are interested in running a   ::
:: Source Dedicated Server or need help, drop by our       ::
:: forums at... http://forums.srcds.com                    ::
::=========================================================::

::=======================::
::  SET YOUR VARIABLES!  ::
::=======================::

::=======================::
::  Window and Log name  ::
::  Replace "My Server"  ::
::=======================::
set servername=Myservername

::=======================::
::   Your start command  ::
::    Replace after =    ::
::=======================::
set runcmd=C:\MYSERVERFOLDERNAME\srcds.exe -game GAMENAME -ip 111.111.1.111 -port 27015 -maxplayers 16 -debug -console -nocrashdialog -norestart -tickrate 66 +map MYMAPNAME

::=======================::
::   End of variables    ::
::=======================::

:: This will keep the window clean and easy to read
@echo off

:: Sets the title of the window
title SRCDS Guardian 3.0   %servername%

:: Clears the window incase there is anything there
cls

:: Prints to the window what we are doing
echo SRCDS Guardian 3.0 has been started!
echo.
echo *************************************************************************
echo To close the server, close this window and type exit in the server window
echo *************************************************************************
echo.
echo.
echo %servername% is now starting...
>> "%servername%.log" echo.
>> "%servername%.log" echo.
>> "%servername%.log" echo (%date%)(%time%) SRCDS Guardian 3.0 has been started!
>> "%servername%.log" echo (%date%)(%time%) %servername% is now starting...

:: This is a return point in case the server crashes or is closed
:restart

echo.
echo (%date%)(%time%) %servername% is now ONLINE
echo Watching %servername% for crashes...
>> "%servername%.log" echo.
>> "%servername%.log" echo (%date%)(%time%) %servername% is now ONLINE
>> "%servername%.log" echo (%date%)(%time%) Watching %servername% for crashes...

::Start the actual server
start /wait %runcmd%

echo.
echo (%date%)(%time%) Crash or Close detected!
echo %servername% is now restarting...
>> "%servername%.log" echo.
>> "%servername%.log" echo (%date%)(%time%) Crash or Close detected!
>> "%servername%.log" echo (%date%)(%time%) %servername% is now restarting...

::Server crashed or closed, so we point it to the return point to start the server again
goto restart`

**Comment by Jpofin** [comment]
Thanks, I got my own way of restarting a server when it crashed and also got kind of a fail2ban system that blocks suspicious IPs, however some people can't connect due to that and if its a large bot-net that attacks you, you're f* anyways..
